server/src/main/java/org/elasticsearch/indices/recovery/RecoverySourceHandler.java

```diff
@@ -315,7 +315,9 @@ && isTargetSameHistory()
                 cancellableThreads,
                 ActionListener.wrap(ignored -> {
                     final long endingSeqNo = shard.seqNoStats().getMaxSeqNo();
-                    logger.trace("snapshot for recovery; current size is [{}]", estimateNumberOfHistoryOperations(startingSeqNo));
+                    if (logger.isTraceEnabled()) {
+                        logger.trace("snapshot for recovery; current size is [{}]", estimateNumberOfHistoryOperations(startingSeqNo));
```

Comment:
nit Maybe more succinct would be `logger.trace(() -> format("snapshot for recovery; current size is [%d]", estimateNumberOfHistoryOperations(startingSeqNo)));` without the "if".

Reply:
But with the allocation of a capturing lambda :P For trace I kinda dislike the lambda version even though it got faster on recent JVMs :)